packages/firebase_auth/firebase_auth_web/lib/firebase_auth_web.dart

```diff
@@ -28,6 +29,29 @@ class FirebaseAuthWeb extends FirebaseAuthPlatform {
     FirebaseAuthPlatform.instance = FirebaseAuthWeb.instance;
     RecaptchaVerifierFactoryPlatform.instance =
         RecaptchaVerifierFactoryWeb.instance;
+
+    FirebaseCoreWeb.setPluginConstantInitializor(
+        'plugins.flutter.io/firebase_auth', constantInitializor);
+  }
+
+  /// Called during initializeApp() to instantiate plugin constants before the user's code
+  /// executes. We listen fr initial auth state event so we're sure web app
+  /// is initialized
+  static Future<Map<String, dynamic>> constantInitializor(app) async {
+    auth_interop.Auth authInstance =
+        auth_interop.getAuthInstance(core_interop.app(app.name));
+    StreamSubscription<auth_interop.User> stream;
+    Completer completer = Completer<Map<String, dynamic>>();
+
+    stream = authInstance.onAuthStateChanged.listen((event) {
+      auth_interop.User user = authInstance.currentUser;
```

Comment:
Does it not make more sense her to use the event provided by the stream? (user or null)